srcpkgs/openssl/template

```diff
@@ -0,0 +1,72 @@
+# Template file for 'openssl'
+pkgname=openssl
+version=1.1.1f
+revision=1
+build_style=configure
+configure_script="./Configure"
+configure_args="--prefix=/usr --openssldir=/etc/ssl --libdir=lib shared
+ no-async no-comp no-idea no-mdc2 no-rc5 no-ec2m no-sm2 no-sm4 no-ssl2
+ no-ssl3 no-seed no-weak-ssl-ciphers -Wa,--noexecstack"
+make_cmd=make
+make_build_args='MAKEDEPPROG="$(CC)'
+make_check_target=test
+hostmakedepends="perl"
+short_desc="Toolkit for Secure Sockets Layer and Transport Layer Security"
+maintainer="John <johnz@posteo.net>"
+license="OpenSSL-License"
+homepage="https://www.openssl.org"
+distfiles="https://www.openssl.org/source/openssl-${version}.tar.gz"
+checksum=186c6bfe6ecfba7a5b48c47f8a1673d0f3b0e5ba2e25602dd23b629975da3f35
+provides="libressl-3.3_${revision}"
+replaces="libressl>=0"
+
+case $XBPS_TARGET_MACHINE in
+	x86_64*) configure_args+=" enable-ec_nistp_64_gcc_128 linux-x86_64";;
+	i686*) configure_args+=" linux-elf";;
+	aarch64*) configure_args+=" linux-aarch64";;
+	ppc64le*) configure_args+=" linux-ppc64le";;
+	ppc64*) configure_args+=" linux-ppc64 no-asm";;
```

Comment:
@q66 might wanna take a look at this, right now it seems to assume ELFv1.

Reply:
Adélie seems to have a patch for this https://code.foxkit.us/adelie/packages/tree/master/system/openssl if you want I can add them here.

Reply:
> Adélie seems to have a patch for this https://code.foxkit.us/adelie/packages/tree/master/system/openssl if you want I can add them here. (Which are also upstream)

Edit: lol, only just noticed that the upstream PR was created by you :D. So I assume you are ware of these patches.

Reply:
yes, just get in those patches and drop the `no-asm`